src/pointer-analysis/value_set_dereference.cpp

```diff
@@ -224,16 +224,14 @@ exprt value_set_dereferencet::dereference(
 
   exprt result_value = nil_exprt{};
 
-  for(std::deque<valuet>::const_iterator it = values.begin();
-      it != values.end();
-      it++)
+  for(const auto &value : values)
   {
-    if(it->value.is_not_nil())
+    if(value.value.is_not_nil())
     {
       if(result_value.is_nil()) // first?
-        result_value = it->value;
+        result_value = value.value;
```

Comment:
`value.value` reads a bit weird in all places. Is there perhaps a better name that could apply? 🤔